kalite/version.py

```diff
@@ -5,386 +9,30 @@
-            "coaches": [],
-            "admins": [],
-        },
-    },
 
-    "0.11.0": {
-        "release_date": "2013/09/03",
-        "git_commit": "29eb96e136d702b5d128b6bbb1b5c347457d080f",
-        "new_features": {
-            "all": [],
-            "students": [],
-            "coaches": [],
-            "admins": ["automated registration", "download and install via zip"],
-        },
-        "bugs_fixed": {
-            "all": [],
-            "students": [],
-            "coaches": [],
-            "admins": [],
-        },
-    },
+def load_yaml(file_name, current_version, yaml_file_version_mapping):
+    """Creates a dictionary from an yaml file.
 
-    "0.10.3": {
-        "release_date": "2014/01/06",
-        "git_commit": "6e56d05e6f53661aea433d72ce7bddacacddc4b8",
-        "new_features": {
-            "all": [],
-            "students": [
-                "Better performance (faster save times and login times)",
-                "Integration with latest exercises",
-            ],
-            "coaches": [],
-            "admins": [
-                "Easier updates for subtitles",
-            ],
-        },
-        "bugs_fixed": {
-            "all": [],
-            "students": [],
-            "coaches": [],
-            "admins": [],
-        },
-    },
+    Args:
+        file_name: The name of the file to be loaded.
+        current_version: The current version of KA Lite.
+        yaml_file_version_mapping: The 'VERSION' string to be replaced by the actual KA Lite version.
 
-    "0.10.2": {
-        "release_date": "2013/10/08",
-        "git_commit": "5831abb8d2ee0815416a17885790679c4672bf97",
-        "new_features": {
-            "all": [],
-            "students": ["Import your KA progress into KA Lite"],
-            "coaches": [],
-            "admins": ["Now start KA Lite by double-clicking the start script"],
-        },
-        "bugs_fixed": {
-            "all": [],
-            "students": [],
-            "coaches": [],
-            "admins": [],
-        },
-    },
+    Returns:
+        A dictionary structure that reflects the yaml structure.
 
-    "0.10.1": {
-        "release_date": "2013/09/16",
-        "git_commit": "f048c11289059f36af0bd1c368f9e0fb354b49b5",
-        "new_features": {
-            "all": ["we have a new (online) website"],
-            "students": [],
-            "coaches": [],
-            "admins": [],
-        },
-        "bugs_fixed": {
-            "all": ["better performance"],
-            "students": [],
-            "coaches": [],
-            "admins": [],
-        },
-    },
+    Raises:
+        IOError: If the file doesn't exist or can't be opened.
+    """
+    try:
+        file_stream = open(file_name, "r")
+        version_object = yaml.load(file_stream)
+        version_object[current_version] = version_object[yaml_file_version_mapping]
+        del version_object[yaml_file_version_mapping]
+        return version_object
+    except IOError:
```

Comment:
Hey @aronasorman !

By "why not just let this bubble up the stack", do you mean, remove de try - except?